apps/memerr/Main.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StaticPointers #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE PatternSynonyms #-}
+
+module Main where
+
+import Control.Distributed.Closure
+import Control.Distributed.Spark as RDD
+import Control.Monad
+import Data.Choice 
+import Data.IORef
+import Data.Text (Text)
+import Options.Applicative as Opt
+
+-- Parser for the command line options
+--
+-- the -n option controls how many references to create 
+-- NOTE: 200,000 is about the number we need to crash the JVM with the 
+-- driver memory set to 512M
+--
+-- and if the --no-io-ref flag is present, we will not store references in an
+-- ioref, so the GHC garbage collecter will have a chance to collect them
```

Comment:
`optparse-applicative` has some combinators that allow to move these comments into the documentation of flags that is displayed with `--help`.